### PR TITLE
CA-554 Strings in postgres preserve case

### PR DESCRIPTION
**Pull request by gpolumbo-broad**
@dvoet - Not sure if we should write unit tests for these things or not.  We don't really have tests that check that we set up the database a particular way, so I wasn't going to start adding them, but what are your thoughts?

Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

**Review by marctalbott — COMMENTED** [review]
Is there a way to do this as a separate changeset? Not sure if you can modify a data type using a Liquibase changeset

File: src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20191021_citext_support.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    <changeSet id="alter_SAM_GROUP_name_to_citext" author="gpolumbo">
+        <modifyDataType tableName="SAM_GROUP" columnName="name" newDataType="citext" />
+    </changeSet>
+
+    <changeSet id="alter_SAM_GROUP_email_to_citext" author="gpolumbo">
+        <modifyDataType tableName="SAM_GROUP" columnName="email" newDataType="citext" />
+    </changeSet>
+
+    <changeSet id="alter_SAM_USER_id_to_citext" author="gpolumbo">
```
Comment (dvoet) [inline src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20191021_citext_support.xml]: do we need to do this one?

File: src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20191021_citext_support.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                                 constraintName="FK_SPSA_USER"
+                                 referencedTableName="SAM_USER"
+                                 referencedColumnNames="id" />
+    </changeSet>
+
+    <changeSet id="alter_SAM_USER_email_to_citext" author="gpolumbo">
+        <modifyDataType tableName="SAM_USER" columnName="email" newDataType="citext" />
+    </changeSet>
+
+    <changeSet id="alter_SAM_PET_SERVICE_ACCOUNT_project_to_citext" author="gpolumbo">
```
Comment (dvoet) [inline src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20191021_citext_support.xml]: google project can only be lower case anyway

**Review by marctalbott — APPROVED** [review]

**Review by dvoet — CHANGES_REQUESTED** [review]

**Comment by dvoet** [comment]
I do think tests would be nice but can follow in another PR if needed

**Pull request merged**